daemon/archive.go

```diff
@@ -375,7 +375,7 @@ func (daemon *Daemon) CopyOnBuild(cID, destPath, srcRoot, srcPath string, decomp
 
 	destExists := true
 	destDir := false
-	rootUID, rootGID := daemon.GetRemappedUIDGID()
+	rootIDs, _ := daemon.idMappings.RootPair()
```

Comment:
The only way for this function to return an error is if the idMaps are non-nil, and they don't have an entry for the root user. I don't know if there is any case where that can happen. 

Maybe it would be better to error out in the `NewIDMappings()` during startup. Instead of checking the same error condition every time it's used. The mapping never changes so it seems like it would make more sense that way.